docs/source/basics/instruct_datasets.rst

```diff
@@ -0,0 +1,262 @@
+    ]
+
+These are then tokenized by the model tokenizer with the appropriate model-specific special tokens added
+(such as beginning-of-sequence, end-of-sequence, and others).
+
+.. code-block:: python
+
+    from torchtune.models.phi3 import phi3_mini_tokenizer
+
+    p_tokenizer = phi3_mini_tokenizer("/tmp/Phi-3-mini-4k-instruct/tokenizer.model")
+    tokens, mask = p_tokenizer.tokenize_messages(msgs)
+    print(tokens)
+    # [1, 32010, 29871, 13, 1792, 9508, 32007, 29871, 13, 32001, 29871, 13, 4299, 2933, 32007, 29871, 13]
+    print(p_tokenizer.decode(tokens))
+    # '\nuser prompt \n \nmodel response \n'
+
+As an example, you can see the schema of the `C4 200M dataset <https://huggingface.co/datasets/liweili/c4_200m>`_.
+
+Loading instruct datasets from Hugging Face
+-------------------------------------------
+
+You simply need to pass in the dataset repo name to ``source``, which is then passed into Hugging Face's ``load_dataset``.
+For most datasets, you will also need to specify the ``split``.
+
+.. code-block:: python
+
+    from torchtune.models.gemma import gemma_tokenizer
+    from torchtune.datasets import instruct_dataset
+
+    g_tokenizer = gemma_tokenizer("/tmp/gemma-7b/tokenizer.model")
+    ds = instruct_dataset(
+        tokenizer=g_tokenizer,
+        source="liweili/c4_200m",
+        split="train"
+    )
+
+.. code-block:: yaml
+
+    # Tokenizer is passed into the dataset in the recipe
+    dataset:
+      _component_: torchtune.datasets.instruct_dataset
+      source: liweili/c4_200m
+      split: train
+
+This will use the default column names "input" and "output". To change the column names, use the ``column_map`` argument.
+
+Loading local and remote instruct datasets
+------------------------------------------
+
+To load in a local or remote dataset via https that follows the instruct format, you need to specify the ``source``, ``data_files`` and ``split``
+arguments. See Hugging Face's ``load_dataset`` `documentation <https://huggingface.co/docs/datasets/main/en/loading#local-and-remote-files>`_
+for more details on loading local or remote files.
+
+.. code-block:: python
+
+    from torchtune.models.gemma import gemma_tokenizer
+    from torchtune.datasets import instruct_dataset
+
+    g_tokenizer = gemma_tokenizer("/tmp/gemma-7b/tokenizer.model")
+    ds = instruct_dataset(
+        tokenizer=g_tokenizer,
+        source="json",
+        data_files="data/my_data.json",
+        split="train",
+    )
+
+.. code-block:: yaml
+
+    # Tokenizer is passed into the dataset in the recipe
+    dataset:
+      _component_: torchtune.datasets.instruct_dataset
+      source: json
+      data_files: data/my_data.json
+      split: train
+
+.. _column_map:
+
+Renaming columns
+----------------
+
+If the default column names are "input", "output" and you need to change them to something else,
+such as "prompt", "response" for example, define ``column_map`` as such:
```

Comment:
Give an explicit of what the raw dataset looks like in this case. The keys and values in this mapping feel backward to me and I think it may be a point of confusion for others